File: scripts/unit_test.py
# Import the functions you want to test here
# Or simply paste them in this script for examples sake - Do not do this in practice!


def some_function(input: str) -> str:
    """Tthe function you want to test."""
    return f"output_{input.split('_')[-1]}"


def test_some_function(
        inputs: list = ["input_1", "input_2", "input_3"],
        expected_outputs: list = ["output_1", "output_2", "output_3"],
) -> None:
    """Modify this function to test your functions."""
    for input, expected_output in zip(inputs, expected_outputs):
        assert some_function(input) == expected_output


def main() -> None:
    """Execute tests."""
    test_some_function()
    print("All tests passed")


if __name__ == "__main__":
    test_some_function()

File: scripts/unit_test.R
# Import the functions you want to test here
# Or simply paste them in this script for examples sake - Do not do this in practice!

some_function <- function(input) {
  parts <- strsplit(input, "_")[[1]]
  return(paste("output", parts[length(parts)], sep = "_"))
}

# Modify this function to test your functions.
test_some_function <- function(inputs = c("input_1", "input_2", "input_3"),
                               expected_outputs = c("output_1", "output_2", "output_3")) {
  for (i in 1:length(inputs)) {
    result <- some_function(inputs[i])
    expected <- expected_outputs[i]
    stopifnot(result == expected)
  }
  print("All awsome tests passed")
}

# Execute the test function
test_some_function()